### PR TITLE
Fix test warnings

### PR DESCRIPTION
**Pull request by hjmjohnson**
Minor updates to allow the test suite to run without creating warnings or errors.

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@tashrifbillah This is a set of small changes that allow the unit tests to pass cleanly.

File: conversion/grad_avg.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def grad_avg(imgFile, outFile, bvalFile= None, bvecFile= None):
         bvecs= read_bvecs(bvecFile)
 
         img= nib.load(imgFile)
-        data = img.get_data()
+        data = img.get_fdata()
```
Comment (tashrifbillah) [inline conversion/grad_avg.py:28]: I have moved to `get_fdata()` in my other projects, did not realize it was still the old one here.

File: conversion/nifti_write.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def nifti_write(inImg, prefix= None):
                 bvec.append(1)
                 # bvecINijk= RAS2IJK @ SPACE2RAS @ mf @ np.matrix(bvec).T
                 # simplified below
-                bvecINijk= xfrm_nhdr.T @ mf @ np.matrix(bvec).T
+                bvecINijk= xfrm_nhdr.T @ mf @ np.array(bvec).T
```
Comment (tashrifbillah) [inline conversion/nifti_write.py:100]: Can you tell the advantage of array over matrix?

File: conversion/grad_avg.py (hunk 1 above)
Comment (hjmjohnson) [reply conversion/grad_avg.py:28]: Just trying to use your good work to save me some time.   :)

I'd love to run the code beautifier 'Black' on the code to make it consistent internally, and consistent with many other medical imaging packages.  Are you supportive of that?

File: conversion/tests/test_file_conversion.py [hunk 3]
```diff
@@ -37,8 +34,11 @@ def test_nrrd2nifti(self):
         testing.assert_array_equal(reference_nifti_data, converted_nifti_data)
         testing.assert_array_equal(reference_nifti_affine, converted_nifti_affine)
         testing.assert_array_equal(reference_nifti_bvals, converted_nifti_bvals)
-        testing.assert_array_equal(reference_nifti_bvecs, converted_nifti_bvecs)
-
+        for bvec_idx in range(0,len(reference_nifti_bvecs)):
+            reference_bvec = array(reference_nifti_bvecs[bvec_idx])
+            converted_bvec = array(converted_nifti_bvecs[bvec_idx])
+            testing.assert_array_almost_equal_nulp(reference_bvec,
+                                                   converted_bvec, 1)
```
Comment (tashrifbillah) [inline conversion/tests/test_file_conversion.py:41]: This is good, you are comparing line by line as opposed to all the lines I compared together before.

File: conversion/grad_avg.py (hunk 1 above)
Comment (tashrifbillah) [reply conversion/grad_avg.py:28]: Code beautifier shouldn't change my algorithm. So feel free to run and propose changes.

File: conversion/tests/test_file_conversion.py (hunk 3 above)
Comment (hjmjohnson) [reply conversion/tests/test_file_conversion.py:41]: I am doing line by line. I could do all the lines together, but then I need to flatten a list of list into a single list.

**Review by tashrifbillah — COMMENTED** [review]
This is my fundamental work. So it is easy for me to review as opposed to UKFTractography which I inherited from my predecessors. I shall pull your changes and do one final test at my machine before merging it. Thank you for contributing Hans. I am all game for making medical imaging packages better.

File: conversion/tests/test_file_conversion.py (hunk 3 above)
Comment (hjmjohnson) [reply conversion/tests/test_file_conversion.py:41]:
```python
#psuedocode
referenence_bvec = []
for bvec_idx  in range(0,len(reference_nifiti_bvecs):
    reference_bvec.extend(  array(reference_nifti_bvecs[bvec_idx]) )
```

File: conversion/nifti_write.py (hunk 2 above)
Comment (hjmjohnson) [reply conversion/nifti_write.py:100]: matrix is deprecated, and there is a long discussion why that is on the numpy website.  the "matrix" class is the same as array, but with a few extra overload functions that are no longer needed now that the '@' operator is available to do the same behavior.

**Comment by hjmjohnson** [comment]
@tashrifbillah Thanks!

**Comment by tashrifbillah** [comment]
```vim
test_nifti2nrrd (test_file_conversion.TestFileConversion) ... Converting  /tmp/conversion/conversion/tests/data/WIP_dti_ax_601.nii.gz
ok
test_nrrd2nifti (test_file_conversion.TestFileConversion) ... ok
test_grad_avg_nifti (test_file_modification.TestFileConversion) ... Total input gradients  16
Averaging gradients:  [2, 3]
Averaging gradients:  [13, 15]
After averaging, total output gradients  14
ok
test_grad_avg_nrrd (test_file_modification.TestFileConversion) ... Total input gradients  16
Averaging gradients:  [13, 15]
/home/tb571/miniconda3/lib/python3.8/site-packages/nrrd/writer.py:323: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.
  raw_data = data.tostring(order=index_order)
After averaging, total output gradients  15
ok
test_grad_remove_nifti_bad_index (test_file_modification.TestFileConversion) ... Reading input ...
Unwanted gradient indices:  [1, 3, 5, 7]
Writing output ...
Writing complete, total output gradients  12
ok
test_grad_remove_nifti_bad_interval (test_file_modification.TestFileConversion) ... Reading input ...
Unwanted gradient indices:  [10, 5]
Writing output ...
Writing complete, total output gradients  14
ok

----------------------------------------------------------------------
Ran 6 tests in 2.946s

OK
```

Tests passed. Shall I go ahead and merge this, @hjmjohnson ?

**Comment by hjmjohnson** [comment]
@tashrifbillah  YES PLEASE!

**Pull request merged**